### DISCUSSION TITLE
JSON Schema Support?

### DISCUSSION THREAD

**Discussion by brophdawg11** [brophdawg11/remix-validity-state#24] [Ideas]
@kentcdodds raised this in https://github.com/brophdawg11/remix-validity-state/issues/21#issuecomment-1287111827

Curious where this might fit into this library (definition or validation) and if it's something that needs first party support or could be provided by utilities (potentially bundled with this library).  Seems like we'd need some ways to go to and from `schema <-> formValidations` as well as a schema validation response and our `serverFormInfo`.

```js
// User defines a schema on their own
let schema = { ... };

// We generate our validations from a schema
// I probably wouldn't want to ship the schema or any complex validation 
// implementations to the client, so this would only do potentially a subset 
// of the validations supported by JSON schema
let validations = generateValidations(schema);

async function action({ request }) {
  let formData = await request.formData();
  // Turn our formData into json
  let json = convertFormDataToJson(formData);
  // Validate our JSON against our schema (instead of using validateServerFormData)
  let schemaValidation = validateAgainstSchema(json, schema);
  // Convert schema validation to serverFormInfo
  let serverFormInfo = convertSchemaValidationToServerFormInfo(schemaValidation);
  return json({ serverFormInfo });
}
```

This would provide:
* JSON-schema-defined `validations``
* A subset of JSON-schema-defined validations handled client-side via `ValidityState`
* Full JSON-schema-validations handled server-side

That seems semi-reasonable - we don't necessarily deal with JSON schema anywhere in the core of this library - but it seems possible to convert to and from in spots as needed?

**Comment by kentcdodds** [comment]
I should probably say that I'm not certain I would opt-into any JSON Schema support. I just brought it up as a "prior art" to potentially look at for inspiration 😅

**Comment by andrecasal** [comment]
I vote for letting this use case be outside the scope of this library (while also keeping it small and to the point 😊)
